### DISCUSSION TITLE
Odd behaviour for autocompletion

### DISCUSSION THREAD

**Discussion by handdara** [latex-lsp/texlab#938] [Q&A]
Context: Ubuntu 22.04, Helix editor running in zellij.

I have a strange error that I wanted to ask about, I feel like it might be just something I'm doing incorrectly. Sometimes when starting commands/environments (e.g. `\begin{equation} ...` ), including user defined "newcommands" the autocomplete searcher will not find anything that doesn't have capital letters. For example, I type a single `\beg` to try to start an environment with `\begin`, but the autocomplete only shows  2 options: `AtBeginDocument` and `AtBeginDvi`. However, if I leave insert mode, and go back and continue typing, suddenly everything shows up.

Any thoughts?

**Comment by pfoerster** [comment]
@handdara It looks like Helix does some client-side filtering. Can you try running creating a log file using `texlab -vvvv --log-file /tmp/texlab.log` and post/upload the section with the `textDocument/completion` request, here (you only need to post the `textDocument/completion` sections if you can't share the document)?
